public/app/core/components/TimelineChart/utils.ts

```diff
@@ -308,22 +310,25 @@ export function prepareTimelineFields(
   for (let frame of series) {
     let startFieldIdx = -1;
     let endFieldIdx = -1;
-
+    let timeFieldCount = 0;
     for (let i = 0; i < frame.fields.length; i++) {
```

Comment:
Do we need to skip over time fields which are hidden here, i.e. `config.custom.hideFrom.viz === true`? I'm not sure I fully understand the intended behavior...